avocado/core/test.py

```diff
@@ -423,6 +424,9 @@ def record_and_warn(*args, **kwargs):
             self.log.debug("  teststmpdir: %s", teststmpdir)
         self.log.debug("  workdir: %s", self.workdir)
 
+        self.__setup_attempt = 0
+        self.__setup_attempts = setup_attempts
```

Comment:
setup_attempt vs setup_attempts can be confusing. The idea is to map, respectively,  "current attempt" and "maximum number of attempts". I propose rename setup_attempts to setup_max_attempts. Likewise the property in avocado.conf and the constant INSTRUMENTED_SETUP_ATTEMPTS.

Reply:
Sounds good.